File: OrdenamientoListaCircular/Persona.cpp
/*******************************************************************************************************
 *            UNIVERSIDAD DE LAS FUERZAS ARMADAS ESPE                                                  *
 * Proposito:                      Clase persona para ordenamiento lista circular                      *
 * Autor:                          Marcelo Acuña, Abner Arboleda, Christian Bonifaz                    *
 * Fecha de creacion:              17/12/2024                                                          *
 * Fecha de modificacion:          18/12/2024                                                          *
 * Materia:                        Estructura de datos                                                 *
 * NRC :                           1992                                                                *
 *******************************************************************************************************/

#include "Persona.h"
#include <iostream>

Persona::Persona(string n, string c, Fecha fn) : nombre(n), cedula(c), fechaNacimiento(fn) {}

string Persona::getNombre() const { return nombre; }

string Persona::getCedula() const { return cedula; }

Fecha Persona::getFechaNacimiento() const { return fechaNacimiento; }

void Persona::setNombre(const string& n) { nombre = n; }

void Persona::setCedula(const string& c) { cedula = c; }

void Persona::setFechaNacimiento(const Fecha& fn) { fechaNacimiento = fn; }

void Persona::mostrar() const {
    cout << "Nombre: " << nombre 
        << ", Cedula: " << cedula 
        << ", Fecha de nacimiento: " << fechaNacimiento.mostrar() << endl;
}

File: Parcial 1 y 2/Listas en python/OrdenamientoListaCircular en python/persona.py
"""
********************************************************************************************
 *            UNIVERSIDAD DE LAS FUERZAS ARMADAS ESPE                                       *
 * Proposito:                      Lista Circular                                           *
 * Autor:                          Abner Arboleda, Marcelo Acuña, Christian bonifaz         *
 * Fecha de creacion:              26/12/2024                                               *
 * Fecha de modificacion:          26/12/2024                                               *
 * Materia:                        Estructura de datos                                      *
 * NRC :                           1992                                                     *
 ********************************************************************************************
"""
from fecha import Fecha
from utilidades import Utilidades

class Persona:
    def __init__(self, nombre="", segundoNombre="", apellido="", cedula="", fechaNacimiento=None, correo=""):
        self.nombre = nombre
        self.segundoNombre = segundoNombre
        self.apellido = apellido
        self.cedula = cedula
        self.fechaNacimiento = fechaNacimiento if fechaNacimiento else Fecha(1, 1, 1970)
        self.correo = correo if correo else Utilidades.generarCorreo(nombre, segundoNombre, apellido)

    def mostrar(self):
        print(f"{self.nombre:<20}{self.segundoNombre:<20}{self.apellido:<15}{self.cedula:<20}{self.correo:<30}{self.fechaNacimiento.strftime('%d-%m-%Y'):<30}")